### PR TITLE
[feature] minimal warehouse support

### PR DESCRIPTION
**Pull request by ryanking**
Only a subset of fields so far, more to come.

### CODE REVIEW

File: pkg/resources/warehouse.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+func ValidateWarehouseName(val interface{}, key string) ([]string, []error) {
+	return snowflake.ValidateIdentifier(val)
+}
+
+func (w *warehouse) Create(data *schema.ResourceData, meta interface{}) error {
+	name := data.Get("name").(string)
+	comment := data.Get("comment").(string)
+	db := meta.(*sql.DB)
+
+	err := DBExec(db, "CREATE WAREHOUSE %s COMMENT='%s", name, comment)
```
Comment (edulop91) [inline pkg/resources/warehouse.go]: this might be missing a `'` around the comment

File: pkg/resources/warehouse.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+func (w *warehouse) Delete(data *schema.ResourceData, meta interface{}) error {
+	db := meta.(*sql.DB)
+	name := data.Get("name").(string)
+
+	err := DBExec(db, "DROP WAREHOUSE %s", name)
+	if err != nil {
+		return errors.Wrapf(err, "error dropping warehouse %s", name)
+	}
+
```
Comment (edulop91) [inline pkg/resources/warehouse.go]: do you also have to `set.ID == ""`? Not sure

File: pkg/resources/warehouse.go (hunk 1 above)
Comment (ryanking) [reply pkg/resources/warehouse.go]: indeed

File: pkg/resources/warehouse.go (hunk 2 above)
Comment (ryanking) [reply pkg/resources/warehouse.go]: yeah I don't know i will look

Comment (ryanking) [reply pkg/resources/warehouse.go]: I don't see any examples of the aws provider doing this, so I assume its not needed.

File: codecov.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+codecov:
```
Comment (edulop91) [inline codecov.yml]: i wonder if we should start a fogg for go repos

File: pkg/resources/warehouse.go [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+	return &warehouse{}
+}
+
+func (w *warehouse) Create(data *schema.ResourceData, meta interface{}) error {
+	db := meta.(*sql.DB)
+	name := data.Get("name").(string)
+
+	var sb strings.Builder
+
+	sb.WriteString(fmt.Sprintf("CREATE WAREHOUSE %s", name))
```
Comment (edulop91) [inline pkg/resources/warehouse.go]: err being ignored

File: pkg/resources/warehouse.go [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+	err = data.Set("comment", comment.String)
+	if err != nil {
+		return err
+	}
+
+	err = data.Set("warehouse_size", size.String)
+	if err != nil {
+		return err
+	}
+	return nil
```
Comment (edulop91) [inline pkg/resources/warehouse.go]: could just return `err` and save a branch

**Pull request merged**